File: solutions/java/rna-transcription/2/src/main/java/RnaTranscription.java
import static java.util.stream.Collectors.joining;

public final class RnaTranscription {

    public static String ofDna(String c) {
        return c.chars()
                .mapToObj(RnaTranscription::transcribe)
                .collect(joining(""));
    }

    private static String transcribe(int c) {
        switch (c) {
            case 'A': return "U";
            case 'C': return "G";
            case 'G': return "C";
            case 'T': return "A";
            default: return "";
        }
    }
}

File: solutions/erlang/rna-transcription/2/rna_transcription.erl
-module(rna_transcription).
-export([to_rna/1]).

-spec to_rna(string()) -> string().
to_rna(Sequence) ->
  lists:map(fun transcribe/1, Sequence).

-spec transcribe(char()) -> char().
transcribe($A) -> $U;
transcribe($C) -> $G;
transcribe($G) -> $C;
transcribe($T) -> $A.